File: Java/Maven/src/main/java/com/COMP3004/Maven/Card.java
package com.COMP3004.Maven;

import java.io.Serializable;

public abstract class Card implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	// Properties
	protected String name, type, deck;

	// Constructor
	public Card(String d, String t, String n) {
		name = n;
		type = t;
		deck = d;
	}

	public String getType() {
		return type;
	}

	public String getName() {
		return name;
	}

	public String getDeck() {
		return deck;
	}

	@Override
	public String toString() {
		return "Card [name=" + name + ", type=" + type + ", deck=" + deck + "]";
	}
	
}

File: Unity/Assets/Scripts/Classes/Deck/Card/Card.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Card: Entity{

    protected string deck;


    //Constructor
    public Card(string n = "", string d = "", playBehaviour p = null):base(n, "CARD", p)
    {
        deck = d;
    }



    public string getDeck()
    {

        return deck;
    }



}

